### PR TITLE
ci: fix luarocks tag release process

### PR DESCRIPTION
**Pull request by teto**
Now tests the rockspec during PRs

### CODE REVIEW

**Comment by amaanq** [comment]
Does this publish the package to luarocks? Might be a better idea to rename the CI action to "Publish"

**Comment by amaanq** [comment]
Ty!

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -1,17 +1,19 @@
 name: "release"
-on:
-  push:
-    tags:
-      - '*'
+on: push
```
Comment (mrcjkb) [inline .github/workflows/release.yml:2]:
```suggestion
on:
  push:
    tags: # Will upload to luarocks.org
      - "*"
  pull_request: # Will test a local install without uploading to luarocks.org
  
```

**Comment by teto** [comment]
should be good now

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -1,17 +1,23 @@
 name: "release"
 on:
   push:
-    tags:
-      - '*'
+    tags: # Will upload to luarocks.org
+      - "*"
+  pull_request: # Will test a local install without uploading to luarocks.org
```
Comment (clason) [inline .github/workflows/release.yml:6]: Is that really necessary?

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -1,17 +1,23 @@
...
           LUAROCKS_API_KEY: ${{ secrets.LUAROCKS_API_KEY }}
         with:
+          name: nvim-treesitter
+          version: 0.1
+          license: apache 2
           detailed_description: |
             The goal of nvim-treesitter is both to provide a simple and easy way to use the interface for tree-sitter in Neovim
             and to provide some basic functionality such as highlighting based on it.
-          build_type: "make"
+          template: nvim-treesitter-luarocks.template
```
Comment (clason) [inline .github/workflows/release.yml:23]: this file should not live on the top level

File: .github/workflows/release.yml (hunk 2 above)
Comment (teto) [reply .github/workflows/release.yml:6]: this allows to check the rockspec doesn't get broken. Some lua packages relies on `luarocks test` to run tests.
What I could do is here is run the action only when the github workflow or template is changed.

File: .github/workflows/release.yml (hunk 3 above)
Comment (teto) [reply .github/workflows/release.yml:23]: sure. Should I create a "contrib" folder and move it there or to "scripts" ?

Comment (clason) [reply .github/workflows/release.yml:23]: I'd prefer if the luarocks files live in `contrib`, yes -- makes their status clear.

File: .github/workflows/release.yml (hunk 2 above)
Comment (clason) [reply .github/workflows/release.yml:6]: Yes, that would be better. Let's start with template only.

**Review by clason — APPROVED** [review]

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -1,17 +1,26 @@
...
     runs-on: ubuntu-latest
     steps:
-      - uses: nvim-neorocks/luarocks-tag-release@v2.2.0
+      - uses: actions/checkout@v3
+      - uses: nvim-neorocks/luarocks-tag-release@v5
         env:
           LUAROCKS_API_KEY: ${{ secrets.LUAROCKS_API_KEY }}
         with:
+          name: nvim-treesitter
+          version: 0.1
```
Comment (mrcjkb) [inline .github/workflows/release.yml:21]: Sorry, I'm a bit late to the party.
Just noticed that the version is fixed here.
Is this left over from testing?

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -1,17 +1,26 @@
...
+          license: apache 2
```
Comment (mrcjkb) [inline .github/workflows/release.yml:22]: The action can fetch the license info from the repo metadata, so this technically isn't needed.

**Pull request merged**